doc/design.md

```diff
@@ -51,10 +51,58 @@ backend --> customerCluster: manages
 @enduml
 ```
 
+#### EM Provides a REST Interface
+`dsn~rest-interface~1`
+
+EM provides a REST interface to clients.
+
+Covers:
+* [`req~rest-interface~1`](system_requirements.md#em-provides-a-rest-interface)
+
+Needs: impl, itest
+
+#### EM Implemented as a Go Library
+`dsn~go-library~1`
+
+EM is implemented as a Go library instead of an executable Go program.
+
+Rationale:
+This allows embedding EM in other applications like SaaS, so it can be reused in different contexts. Nevertheless EM additionally contains an executable Go program which is useful for testing EM and extensions.
+
+Covers:
+* [`req~embeddable-rest-interface~1`](system_requirements.md#rest-interface-is-embeddable)
+
+Needs: impl
+
+#### EM Generates an OpenAPI Specification
+`dsn~openapi-spec~1`
+
+EM generates an OpenAPI specification at runtime using the [Nightapes/go-rest](https://github.com/Nightapes/go-rest) library.
+
+The standalone executable of EM provides the OpenAPI specification as a web page and in JSON format. Applications that embed EM as a library need to create their own endpoint for that.
+
+Covers:
+* [`req~openapi-spec~1`](system_requirements.md#em-provides-an-openapi-specification)
+ 
+Needs: impl, itest
+
+#### Extension Registry
+`dsn~extension-registry~1`
+
+EM uses a web service called Extension Registry (similar to service discovery) to find available extensions and their JavaScript definitions.
+
+Rationale:
+
+This will allow updating extension definitions without modifying the deployed EM. In the future this will also allow multiple instances of EM to use the same list of available extensions. During development and for integration tests it is easier to use a local directory with JavaScript files for finding available extensions. That's why EM supports both variants. At startup it will check if the configured extension registry URL starts with `http` and uses the appropriate implementation.
```

Comment:
```suggestion
This will allow updating extension definitions without modifying the deployed EM. In the future this will also allow multiple instances of EM to use the same list of available extensions. During development and for integration tests it is easier to use a local directory with JavaScript files for finding available extensions. That's why EM supports both variants. At startup it will check if the configured extension registry URL starts with `http` and will use the appropriate implementation.
```

Reply:
Don't use future. We assume that at the time the design is released, each written down requirement is already covered.